src/mlpack/methods/amf/init_rules/given_init.hpp

```diff
@@ -28,32 +28,89 @@ class GivenInitialization
 {
  public:
   // Empty constructor required for the InitializeRule template.
-  GivenInitialization() { }
+  GivenInitialization() : wIsGiven(false), hIsGiven(false) { }
 
   // Initialize the GivenInitialization object with the given matrices.
-  GivenInitialization(const arma::mat& w, const arma::mat& h) : w(w), h(h) { }
+  GivenInitialization(const arma::mat& w, const arma::mat& h) :
+    w(w), h(h), wIsGiven(true), hIsGiven(true) { }
 
   // Initialize the GivenInitialization object, taking control of the given
   // matrices.
   GivenInitialization(const arma::mat&& w, const arma::mat&& h) :
     w(std::move(w)),
-    h(std::move(h))
+    h(std::move(h)),
+    wIsGiven(true),
+    hIsGiven(true)
   { }
 
+  // Initialize either H or W with the given matrix.
+  GivenInitialization(const char whichMatrix, const arma::mat& m)
```

Comment:
I guess we could reuse the other function.

Reply:
Can you be more specific? Do you mean I can merge the two constructors `GivenInitialization(const char whichMatrix, const arma::mat& m)` and `GivenInitialization(const char whichMatrix, const arma::mat&& m)` into one?